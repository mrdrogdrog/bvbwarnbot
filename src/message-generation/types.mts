import { DateTime } from "luxon";


export type Reason = "emGermanTeam" | "emLocationDortmund" | "bvbHomecoming" | "bvbAway"


export interface MatchData {
  homeTeam: string,
  awayTeam: string,
  time: DateTime
  reason: Reason
}
