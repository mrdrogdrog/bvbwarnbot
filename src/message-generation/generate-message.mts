import { DateTime } from "luxon";
import { renderFile } from "ejs";


export async function generateMessage(hour: number, date: DateTime, awayTeam: string, homeTeam: string, homecoming: boolean): Promise<string> {
  const formattedDate = `${toRelativeTime(date)}${date.toFormat("dd.MM.yyyy - HH:mm")}`;

  return await renderFile("./message-template.ejs", { formattedDate, awayTeam, homeTeam, homecoming, hour });
}

function toRelativeTime(date: DateTime) {
  const now = DateTime.now();
  if (date.day === now.day) {
    return "Heute, ";
  } else if (date.day === now.day + 1) {
    return "Morgen, ";
  } else {
    return "";
  }
}
