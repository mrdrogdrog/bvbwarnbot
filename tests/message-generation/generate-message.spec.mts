import { DateTime, Settings, Zone } from "luxon";
import { generateMessage } from "../../src/message-generation/generate-message.mjs";
import { MatchData } from "../../src/message-generation/types.mjs";
import { expect, describe, it, vi, beforeEach } from 'vitest'

describe("message template", () => {
  beforeEach(() => {
    Settings.now = () => 0;
    Settings.defaultZone = "Europe/Berlin";
  });

  it("generates a bvb homecoming message", async () => {
    const match: MatchData = {
      time: DateTime.fromSeconds(0),
      awayTeam: "AWAY",
      homeTeam: "HOME",
      reason: "bvbHomecoming"
    };
    const message = await generateMessage(123, match);
    expect(message).toMatchSnapshot();
  });

  it("generates a bvb away message", async () => {
    const match: MatchData = {
      time: DateTime.fromSeconds(0),
      awayTeam: "HOME",
      homeTeam: "AWAY",
      reason: "bvbAway"
    };
    const message = await generateMessage(123, match);

    expect(message).toMatchSnapshot();
  });

  it("generates an EM german team message", async () => {
    const match: MatchData = {
      time: DateTime.fromSeconds(0),
      awayTeam: "HOME",
      homeTeam: "AWAY",
      reason: "emGermanTeam"
    };
    const message = await generateMessage(123, match);

    expect(message).toMatchSnapshot();
  });

  it("generates an EM Dortmund Location message", async () => {
    const match: MatchData = {
      time: DateTime.fromSeconds(0),
      awayTeam: "HOME",
      homeTeam: "AWAY",
      reason: "emLocationDortmund"
    };
    const message = await generateMessage(123, match);

    expect(message).toMatchSnapshot();
  });

  it("generates an EM German Team in Dortmund Location message", async () => {
    const match: MatchData = {
      time: DateTime.fromSeconds(0),
      awayTeam: "HOME",
      homeTeam: "AWAY",
      reason: "emGermanTeamInDortmund"
    };
    const message = await generateMessage(123, match);

    expect(message).toMatchSnapshot();
  });

  it("generates a message for tomorrow", async () => {
    const match: MatchData = {
      time: DateTime.fromSeconds(100000),
      awayTeam: "HOME",
      homeTeam: "AWAY",
      reason: "bvbAway"
    };

    const message = await generateMessage(123, match);

    expect(message).toMatchSnapshot();
  });


  it("generates a message for more than tomorrow", async () => {
    const match: MatchData = {
      time: DateTime.fromSeconds(100000000),
      awayTeam: "HOME",
      homeTeam: "AWAY",
      reason: "bvbAway"
    };

    const message = await generateMessage(123, match);

    expect(message).toMatchSnapshot();
  });
});

