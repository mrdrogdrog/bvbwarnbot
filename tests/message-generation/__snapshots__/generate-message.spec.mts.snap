// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message template generates a bvb away message 1`] = `
"⚠️123h-Warnung!⚠️

Heute, 01.01.1970 - 01:00

AWAY vs HOME

🏕BVB-Auswärtsspiel

Vermeidet Kneipen mit TV
"
`;

exports[`message template generates a bvb homecoming message 1`] = `
"⚠️123h-Warnung!⚠️

Heute, 01.01.1970 - 01:00

HOME vs AWAY

🏠BVB-Heimspiel

Vermeidet Innenstadt / U42 / U46 / Kreuzviertel / Borsigplatz / Uni-Parkplatz
"
`;

exports[`message template generates a message for more than tomorrow 1`] = `
"⚠️123h-Warnung!⚠️

03.03.1973 - 10:46

AWAY vs HOME

🏕BVB-Auswärtsspiel

Vermeidet Kneipen mit TV
"
`;

exports[`message template generates a message for tomorrow 1`] = `
"⚠️123h-Warnung!⚠️

Morgen, 02.01.1970 - 04:46

AWAY vs HOME

🏕BVB-Auswärtsspiel

Vermeidet Kneipen mit TV
"
`;

exports[`message template generates an EM Dortmund Location message 1`] = `
"⚠️123h-Warnung!⚠️

Heute, 01.01.1970 - 01:00

AWAY vs HOME

🇪🇺EM-Spiel
🏟️Dieses Spiel findet in Dortmund statt

Vermeidet Innenstadt / U42 / U46 / Kreuzviertel / Borsigplatz / Uni-Parkplatz
"
`;

exports[`message template generates an EM German Team in Dortmund Location message 1`] = `
"⚠️123h-Warnung!⚠️

Heute, 01.01.1970 - 01:00

AWAY vs HOME

🇪🇺EM-Spiel
🇩🇪Spiel des deutschen Team
🏟️Dieses Spiel findet in Dortmund statt

Vermeidet die Innenstadt und Kneipen mit TV.
Am besten einfach nicht raus gehen.
Lasst alle Hoffnungen fahren und verbarrikadiert euch.
Es ist ein Deutschland-Spiel IN Dortmund 😱😱😱
"
`;

exports[`message template generates an EM german team message 1`] = `
"⚠️123h-Warnung!⚠️

Heute, 01.01.1970 - 01:00

AWAY vs HOME

🇪🇺EM-Spiel
🇩🇪Spiel des deutschen Team

Vermeidet die Innenstadt und Kneipen mit TV!
"
`;
